PurchasesCoreSwift/Public/Error Handling/ErrorUtils.swift

```diff
@@ -0,0 +1,291 @@
+//
+//  PurchasesErrorUtils.swift
+//  PurchasesCoreSwift
+//
+//  Created by César de la Vega on 7/19/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+@objc(RCPurchasesErrorUtils) public class ErrorUtils: NSObject {
+
+    /**
+     * Constructs an NSError with the [RCNetworkError] code and a populated [RCUnderlyingErrorKey] in
+     * the [NSError.userInfo] dictionary.
+     *
+     * @param underlyingError The value of the [NSUnderlyingErrorKey] key.
+     *
+     * @note This error is used when there is an error performing network request returns an error or when there
+     * is an [NSJSONSerialization] error.
+     */
+    @objc public static func networkError(withUnderlyingError underlyingError: Error) -> Error {
```

Comment:
I am not sure if these should return `ErrorCode` instead, to indicate that's one of those or `Error` is fine.